client/http/client.go

```diff
@@ -169,11 +185,13 @@ func (ci *clientInner) requestWithRetry(
 		addr = ci.pdAddrs[idx]
 		err = ci.doRequest(ctx, addr, reqInfo, headerOpts...)
 		if err == nil {
-			break
+			return nil
 		}
 		log.Debug("[pd] request follower addr failed",
 			zap.String("source", ci.source), zap.Int("idx", idx), zap.String("addr", addr), zap.Error(err))
 	}
+	// Schedule the members info update if all the requests failed to get the latest members as soon as possible.
+	ci.scheduleUpdateMembersInfo()
```

Comment:
better using retry mechanism

Reply:
After using ServiceDiscovery, we can use `ScheduleCheckMemberChanged`. How about waiting for #7668